### DISCUSSION TITLE
Trigger updating details row when only master record is modified

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#6965] [Help] [answered]
I have a master dialog and a details grid that saving and updating works perfectly as long as at least one of the details record is also modified.
But this doesn't work if only the master record is modified.
There is a value in the details row that is dependent on a value in the master row, if the master row record is modified it should modify that in the details row but it doesn't.

My question is how do you trigger the updating of the details record when only the master record is modified.

**Comment by reach2rv** [comment, marked as answer]
On chanheselect2 of the master field. Iterate through detail items through for each loop and update / set relevant value. Then on save it will update detail records also.

**Reply by rhemasystems** [reply]
Thanks man, it worked

`
this.form.TenantId.change(e => {

    this.tenantid = Q.toId(this.form.TenantId.value);
    var criteria: any[];

    if (this.tenantid != null) {
        if (this.form.LineItems.rowCount() > 0) {
            for (var k of this.form.LineItems.getItems()) {
                k.TenantId = this.tenantid
            }
        }
    };
});
`
